cmd/up/exec.go

```diff
@@ -16,27 +16,268 @@ package up
 import (
 	"bytes"
 	"context"
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
+	"io"
 	"os"
+	"os/exec"
+	"path/filepath"
 	"strings"
 
+	"github.com/creack/pty"
 	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
 	"github.com/okteto/okteto/pkg/config"
+	"github.com/okteto/okteto/pkg/constants"
 	oktetoErrors "github.com/okteto/okteto/pkg/errors"
 	"github.com/okteto/okteto/pkg/k8s/apps"
-	"github.com/okteto/okteto/pkg/k8s/exec"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	k8sExec "github.com/okteto/okteto/pkg/k8s/exec"
 	"github.com/okteto/okteto/pkg/k8s/pods"
 	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/registry"
 	"github.com/okteto/okteto/pkg/ssh"
+	"github.com/okteto/okteto/pkg/types"
+	"k8s.io/client-go/kubernetes"
 )
 
+type devExecutor interface {
+	runCommand(ctx context.Context, cmd []string) error
+}
+
+type hybridExecutor struct {
+	workdir string
+	envs    []string
+}
+
+func (he *hybridExecutor) runCommand(_ context.Context, cmd []string) error {
+	c := exec.Command("bash", "-c", strings.Join(cmd, " "))
+	c.Dir = he.workdir
+	c.Env = append(os.Environ(), he.envs...)
```

Comment:
We shouldn't be adding the os.Environ variables. We need to be as close as it would be running on the cloud. I'd add only needed environment variables such as `PATH` but definetely not adding all of them since we are retrieving them from the cluster

Reply:
done 9c398bf87ad1cf49cfdde25f350c7da00c2b2671